geopandas/base.py

```diff
@@ -374,7 +374,7 @@ def is_ring(self):
         >>> s
         0    LINESTRING (0.00000 0.00000, 1.00000 1.00000, ...
         1    LINESTRING (0.00000 0.00000, 1.00000 1.00000, ...
-        2    LINEARRING (0.00000 0.00000, 1.00000 1.00000, ...
+        2    LINESTRING (0.00000 0.00000, 1.00000 1.00000, ...
```

Comment:
This is not correct ... it should be LINEARRING (since that is what is being created)

But that seems to be an issue with the shapely->pygeos conversion. I find it strange that it comes up now, though. 

When going through WKB, this can be expected (since LinearRing doesn't exist in WKB). But normally here on CI we are using conda, and then pygeos should clone the underlying GEOS geometry directly instead of going through WK.

Reply:
OK, found it: it is not because of the shapely->pygeos conversion, but actually from the pygeos->shapely conversion that happens at the moment we print the values (the repr works by getting each element, and formatting that element with shapely.wkt.dumps). Getting scalar elements from a GeoSeries/GeometryArray calls `_pygeos_to_shapely`, where we have:

https://github.com/geopandas/geopandas/blob/f17b5350b8fa8acb8926e78ee011f876c7a03ff4/geopandas/_vectorized.py#L71-L87

So the "compatible fast path" (which would preserve the exact LinearRing geometry type) is no longer used if have both pygeos and shapely 2.0

For the fallback through WKB, we could add a special case for LinearRing to preserve the type ..

Reply:
Thanks for working out why this was happening (and also for the detailed write up on the shapely repo, that was interesting to know more about), would have taken me a while to work out. I've pushed a fix for this and added a test so that we cover this outside of the doctests / the specific environment doctests are run on.